src/block_storage/volumes.rs

```diff
@@ -0,0 +1,453 @@
+        links: ref Vec<protocol::Link>
+    }
+
+    transparent_property! {
+        #[doc = "Name of the availability zone."]
+        availability_zone: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Current backend of the volume."]
+        host: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Whether the volume is encrypted."]
+        encrypted: ref bool
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the encryption key."]
+        encryption_key_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "When the volume was last updated."]
+        updated_at: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Volume replication status."]
+        replication_status: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the snapshot the volume originated from."]
+        snapshot_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the volume."]
+        id: ref String
+    }
+
+    transparent_property! {
+        #[doc = "Size of the volume in GiB."]
+        size: ref u64
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the user."]
+        user_id: ref String
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the project."]
+        tenant_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Migration status."]
+        migstat: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Metadata of the volume."]
+        metadata: ref HashMap<String, String>
+    }
+
+    transparent_property! {
+        #[doc = "Status of the volume."]
+        status: ref protocol::VolumeStatus
+    }
+
+    transparent_property! {
+        #[doc = "Metadata of the image used to create the volume."]
+        image_metadata: ref Option<HashMap<String, String>>
+    }
+
+    transparent_property! {
+        #[doc = "Description of the volume."]
+        description: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Whether the volume is multi-attachable."]
+        multi_attachable: ref bool
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the volume this one originated from."]
+        source_volume_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the consistency group."]
+        consistency_group_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the volume that this volume name on the backend is based on."]
+        name_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Name of the volume."]
+        name: ref String
+    }
+
+    transparent_property! {
+        #[doc = "Whether the volume is bootable."]
+        bootable: ref String
+    }
+
+    transparent_property! {
+        #[doc = "When the volume was created."]
+        created_at: ref String
```

Comment:
Would it be possible to convert this to DateTime automatically? You can check other resources for examples

Reply:
So, I thought about this and think I might have a solution. I created an enum `block_storage::protocol::DateTime` which wraps `chrono::DateTime<FixedOffset>` and `chrono::NaiveDateTime` in its variants `WithTz` and `WithoutTz` and implemented `Deserialize` accordingly. I also added other `impl`s that the `protocol_enum` would usually add.

This now works with our Zed cluster here where I don't seem to get timezone information, but would also work on clusters where you would.